scripts/xtensa-build-zephyr.py

```diff
@@ -374,6 +376,57 @@ def get_sof_version(abs_build_dir):
 		      versions['SOF_MICRO']
 	return sof_version
 
+def get_overlays(plt_dict):
+		overlays = [str(item.resolve(True)) for item in args.overlay]
+		# You may override default overlay.conf file name using CONFIG_OVERLAY in your platform
+		# dictionary
+		platform = plt_dict.get('name')
+		overlay_filename = plt_dict.get("CONFIG_OVERLAY", "overlay.conf")
+		overlays.append(str(pathlib.Path(SOF_TOP, "overlays", platform, overlay_filename)))
+
+		# The '-d' option is a shortcut for '-o path_to_debug_overlay', we are good
+		# if both are provided, because it's no harm to merge the same overlay twice.
+		if args.debug:
+			overlays.append(str(pathlib.Path(SOF_TOP, "overlays", "common", "debug_overlay.conf")))
+
+		# The '-i IPC4' is a shortcut for '-o path_to_ipc4_overlay', we are good if both
+		# are provided, because it's no harm to merge the same overlay twice.
+		if args.ipc == "IPC4":
+			overlays.append(str(pathlib.Path(SOF_TOP, "overlays", platform, plt_dict["IPC4_CONFIG_OVERLAY"])))
+
+		overlays = ";".join(overlays)
+		return overlays
+
+def sparse_build():
+	# Need to build sparse from source in order for it to support the xtensa architecture:
+	# git clone "git clone https://git.kernel.org/pub/scm/devel/sparse/sparse.git/"
+	# cd sparse; make && make install
+	#
+	# Set cgcc to Zephyr SDK gcc as the compiler to invoke:
+	# export REAL_CC=/$YOUR_PATH/zephyr-sdk-$SDK_VERSION/xtensa-intel_s1000_zephyr-elf/bin/xtensa-intel_s1000_zephyr-elf-gcc
+	global west_top, SOF_TOP
+	print(f"SOF_TOP={SOF_TOP}")
+	print(f"west_top={west_top}")
+
+	assert(west_top.exists())
+	for platform in args.platforms:
+		platform_dict = [x for x in platform_list if x["name"] == platform][0]
+		PLAT_CONFIG = platform_dict["PLAT_CONFIG"]
+		sparse_build_cmd = ["west"]
+		sparse_build_cmd += ["-v"] * args.verbose
+		sparse_build_cmd += ["build", "-d", platform + '-sparse']
+		source_dir = pathlib.Path(west_top, "zephyr", "samples", "subsys", "audio", "sof")
+		sparse_build_cmd += ["--board", PLAT_CONFIG, str(source_dir)]
+		sparse_build_cmd.append('--')
+		if args.cmake_args:
+			build_cmd += args.cmake_args
+		sparse_build_cmd += ["-DSPARSE=y"]
```

Comment:
Is it possible to just pass `-DSPARSE=y` to `cmake_args` in the command line when we want Sparse build.

Reply:
@lyakh  I assume Zephyr has a build option to enable sparse at the top level we should reuse here ? Can you comment.

Reply:
`west` (= Zephyr's "top-level" has the ability to passthrough anything to CMake, which is what @keqiaozhang did here: https://docs.zephyrproject.org/latest/develop/west/build-flash-debug.html#one-time-cmake-arguments

This is infinitely scalable and saves `west` the need to add a new `west --sparse` option and then a new `west --something` option and then `west --something-else` etc.


I think `west build ... -- too_long_list_of_overalds -DSPARSE=y` is correct, I also think it would be much less code in a semi-hardcoded CI script. No need to add it to this python script (that the Zephyr team does not like)

In the future `too_long_list_of_overlays` should be gone and usage of this script reduced.

Reply:
@marc-hb ok, thanks - topic for Wednesday call.

Reply:
> @lyakh I assume Zephyr has a build option to enable sparse at the top level we should reuse here ? Can you comment.

@lgirdwood sorry, I thought I replied to this: that is the correct way to build with sparse support - by adding the `-DSPARSE=y` key to cmake